Wrappers/Python/cil/io/TIFF.py

```diff
@@ -29,40 +29,69 @@
 import glob
 import re
 import numpy as np
+from cil.io import utilities
 
 import logging
 
 logger = logging.getLogger(__name__)
 
+def compress_data(data, scale, offset, dtype):
+    '''Compress data to dtype using scale and offset
+    
+    Parameters
+    ----------
+    data : numpy array
+    scale : float
+    offset : float
+    dtype : numpy dtype
+    
+    returns compressed casted data'''
+    if dtype == data.dtype:
+        return data
+    if data.ndim > 2:
+        # compress each slice
+        tmp = np.empty(data.shape, dtype=dtype)
+        for i in range(data.shape[0]):
+            tmp[i] = compress_data(data[i], scale, offset, dtype)
+    else:
+        tmp = data * scale + offset
+        tmp = tmp.astype(dtype)
+    return tmp
+
 class TIFFWriter(object):
     '''Write a DataSet to disk as a TIFF file or stack'''
     
     def __init__(self,
                  **kwargs):
         '''
-        :param data: the data to save to TIFF file(s)
-        :type data: DataContainer, AcquisitionData or ImageData
-        :param file_name: string defining the file name prefix
-        :type file_name: string
-        :param counter_offset: int indicating at which number the index starts. 
+
+        Parameters
+        ----------
+        data : the DataContainer, AcquisitionData or ImageData to save to TIFF file(s)
+        file_name : string defining the file name prefix
+        counter_offset : int indicating at which number the ordinal index should start. Default 0.
           For instance, if you have to save 10 files the index would by default go from 0 to 9.
           By counter_offset you can offset the index: from `counter_offset` to `9+counter_offset`
-        :type counter_offset: int, default 0
+        compression : The lossy compression to apply, default 0 will not compress data. 
+          8 or 16 will compress to 8 and 16 bit dtypes respectively, default 0.
```

Comment:
```suggestion
        compression : The lossy compression to apply, default 0 will not compress data. 
          8 or 16 will compress to 8 and 16 bit dtypes respectively.
```
Mentioned default value twice

Reply:
I like the default to be at the last place, so I can find it quickly.

Reply:
ok